plugins/device/transaction.go

```diff
@@ -1,52 +1,50 @@
 package device
 
 import (
-	"os/exec"
 	"strings"
 
 	"github.com/gluster/glusterd2/glusterd2/transaction"
 	deviceapi "github.com/gluster/glusterd2/plugins/device/api"
-
-	"github.com/pborman/uuid"
-	log "github.com/sirupsen/logrus"
+	"github.com/gluster/glusterd2/plugins/device/deviceutils"
 )
 
 func txnPrepareDevice(c transaction.TxnCtx) error {
-	var peerID uuid.UUID
-	var req deviceapi.AddDeviceReq
-	var deviceList []deviceapi.Info
-	if err := c.Get("peerid", peerID); err != nil {
-		c.Logger().WithError(err).Error("Failed transaction, cannot find peer-id")
+	var peerID string
+	if err := c.Get("peerid", &peerID); err != nil {
+		c.Logger().WithError(err).WithField("key", "peerid").Error("Failed to get key from transaction context")
 		return err
 	}
-	if err := c.Get("req", req); err != nil {
-		c.Logger().WithError(err).Error("Failed transaction, cannot find device-details")
+
```

Comment:
Suggestion for improvement (not in this PR but later):

Every call to `c.Get` translates to a etcd client call. You can create a local structure containing both peer ID and device name and then set it in the transaction context.